Increase version to 10.1 prealpha

PVince81 opened the pull request:
Set version to 10.1 prealpha.

Assuming we still allow to upgrade from 9.0.9 thanks to all the migration code all the way up to 10.1.

@DeepDiver1975 @phisch @jvillafanez

Commit:
Increase version to 10.1 prealpha

PVince81 commented:
Note: most marketplace apps not compatible:
```
Repair warning: please install app manually with tarball or disable them with:
occ app:disable comments
occ app:disable dav
occ app:disable federatedfilesharing
occ app:disable federation
occ app:disable files
occ app:disable files_external
occ app:disable files_sharing
occ app:disable files_trashbin
occ app:disable files_versions
occ app:disable provisioning_api
occ app:disable systemtags
occ app:disable updatenotification
```

PVince81 commented:
should we the wait before increasing version ?

DeepDiver1975 commented:
> should we the wait before increasing version ?

well - we need to bump these version as well ....

PVince81 commented:
bump the versions first and unbundle later ? or unbundle first ?

PVince81 commented:
okay of course I'll update the min version of these bundled apps...

PVince81 commented:
Here we go. With only git there is no more error on `occ upgrade`.

Repository apps (ex: activity, etc) will also need version adjustements.

PVince81 commented:
Merge this or wait a bit longer ? (might block dev flow until we adjust all apps)

DeepDiver1975 commented:
Let's wait a bit until all apps are in marketplace

PVince81 commented:
- [ ] make PHP 7.1 the minimum in composer.json

PVince81 commented:
@DeepDiver1975 move forward with this ? if apps are not compatible then so be it. they need to be made compatible and be tested one by one and released to the marketplace.

The moment where all apps must be in the marketplace and compatible is *before* 10.1 RC1

DeepDiver1975 commented:
yes - lets merge this in .... things need to move on

Commit:
Set min PHP version to 7.1

PVince81 commented:
rebased

also updated min php version to php 7.1

Once merged let's go with the Guzzle update as well. We'll need to update individual apps to be compatible with this (fresh) by updating their guzzle libs as well.

It also means such apps will need two branches: one for 10.0 and one for 10.1.

PVince81 commented:
@patrickjahns as I understand updating phpunit to 6 would break out tests, so this will need to be done separately with the backportable and forward-compatible namespace changes

phil-davis commented:
can we merge #30242 first?
to get a "nice" point of being up-to-date with composer versions against PHP 5.6
(that "backports" to stable10 nicely and gives a point of comparison between master and stable10 before versions of stuff all diverge)

PVince81 commented:
Fine by me.

- [ ] merge https://github.com/owncloud/core/pull/30242 first

@DeepDiver1975 please review ^

DeepDiver1975 commented:
@ownclouders rebase

ownclouders commented:
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

ownclouders commented:
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

DeepDiver1975 committed:
Fix Jenkinsfile

DeepDiver1975 committed:
Fix drone config

DeepDiver1975 committed:
PHP 7.1+ is required

phil-davis commented:
Another little (last) composer change that could go in before this PHP 7.1 bump:

- [ ] merge #30401 also

@DeepDiver1975 please review ^

DeepDiver1975 committed:
Jenkinsfile: use php 7.1.0

DeepDiver1975 commented:
LOL - jenkins is so nuts

```
15:46:33 This version of ownCloud requires at least PHP 7.1.0
15:46:33 You are currently running PHP 7.1.0RC4. Please update your PHP version.
15:46:33 No coverage
```

I need to update php on that machine ......

DeepDiver1975 commented:
WTF? https://jenkins.owncloud.org/job/owncloud-core/job/core/job/PR-28043/8/console

```
16:23:32 .........................................................PHP Fatal error:  Class Mock_DriverException_5eb04368 cannot implement interface Throwable, extend Exception or Error instead in /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit-mock-objects/src/Framework/MockObject/Generator.php(345) : eval()'d code on line 1
16:23:32 PHP Stack trace:
16:23:32 PHP   1. {main}() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/phpunit:0
16:23:32 PHP   2. PHPUnit_TextUI_Command::main() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/phpunit:52
16:23:32 PHP   3. PHPUnit_TextUI_Command->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/TextUI/Command.php:116
16:23:32 PHP   4. PHPUnit_TextUI_TestRunner->doRun() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/TextUI/Command.php:186
16:23:32 PHP   5. PHPUnit_Framework_TestSuite->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/TextUI/TestRunner.php:517
16:23:32 PHP   6. PHPUnit_Framework_TestSuite->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestSuite.php:733
16:23:32 PHP   7. PHPUnit_Framework_TestCase->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestSuite.php:733
16:23:32 PHP   8. PHPUnit_Framework_TestResult->run() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestCase.php:868
16:23:32 PHP   9. PHPUnit_Framework_TestCase->runBare() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestResult.php:686
16:23:32 PHP  10. PHPUnit_Framework_TestCase->runTest() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestCase.php:913
16:23:32 PHP  11. ReflectionMethod->invokeArgs() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestCase.php:1062
16:23:32 PHP  12. Test\DB\AdapterTest->testUpsertCatchDeadlockAndThrowsException() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestCase.php:1062
16:23:32 PHP  13. PHPUnit_Framework_TestCase->createMock() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/tests/lib/DB/AdapterTest.php:135
16:23:32 PHP  14. PHPUnit_Framework_MockObject_MockBuilder->getMock() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit/src/Framework/TestCase.php:1466
16:23:32 PHP  15. PHPUnit_Framework_MockObject_Generator->getMock() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit-mock-objects/src/Framework/MockObject/MockBuilder.php:117
16:23:32 PHP  16. PHPUnit_Framework_MockObject_Generator->getObject() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit-mock-objects/src/Framework/MockObject/Generator.php:268
16:23:32 PHP  17. PHPUnit_Framework_MockObject_Generator->evalClass() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit-mock-objects/src/Framework/MockObject/Generator.php:292
16:23:32 PHP  18. eval() /var/lib/jenkins/workspace/owncloud-core_core_PR-28043-BOJGKI6NLEZIKS5M6MYWFIBVGMTPQC6S4JKWY3BZKO5MWC5SLI7A/lib/composer/phpunit/phpunit-mock-objects/src/Framework/MockObject/Generator.php:345
16:23:32 Kill the swift docker
16:23:32 Docker executable found - stop and remove docker containers
16:23:32 Stopping and removing docker container 7976700a7a2b4e3d903193e62c36fa52a3cb3de154d49b7568a83f774f8cd14d
16:23:44 7976700a7a2b4e3d903193e62c36fa52a3cb3de154d49b7568a83f774f8cd14d
16:23:44 7976700a7a2b4e3d903193e62c36fa52a3cb3de154d49b7568a83f774f8cd14d
16:23:44 Makefile:163: recipe for target 'test-php' failed
16:23:44 make: *** [test-php] Error 255

patrickjahns commented:
If we continue with php7.1 > we should adjust .drone.yml to only run php7.1 and php7.2 and drop php5.6 & 7.0

DeepDiver1975 commented:
> If we continue with php7.1 > we should adjust .drone.yml to only run php7.1 and php7.2 and drop php5.6 & 7.0

done in commits above - there is an issue in mocking interfaces which inherit from \Throwable - doctrine/dbal 2.6 does so ...

phil-davis commented:
And ``.travis.yml`` s/5.6/7.1/ for all those UI test jobs...

DeepDiver1975 commented:
> there is an issue in mocking interfaces which inherit from \Throwable - doctrine/dbal 2.6 does so ...

https://github.com/sebastianbergmann/phpunit/issues/2995

DeepDiver1975 committed:
Adjust travis.yml

DeepDiver1975 commented:
> And .travis.yml s/5.6/7.1/ for all those UI test jobs...

done

DeepDiver1975 commented:
> there is an issue in mocking interfaces which inherit from \Throwable - doctrine/dbal 2.6 does so ...
> 
> sebastianbergmann/phpunit#2995

fix is there - https://github.com/sebastianbergmann/phpunit-mock-objects/pull/402

DeepDiver1975 committed:
[Workaround] .... until https://github.com/sebastianbergmann/phpunit-mock-objects/pull/402 is merged and released

DeepDiver1975 committed:
travis php 7.0 -> 7.1

DeepDiver1975 commented:
blocked by https://github.com/owncloud-ci/php/issues/34

DeepDiver1975 commented:
@phil-davis @individual-it mind having a look at this https://drone.owncloud.com/owncloud/core/3775/90

![bildschirmfoto von 2018-02-12 17-57-29](https://user-images.githubusercontent.com/1005065/36108938-376e3d32-101e-11e8-8e1a-46047e14474a.png)

DeepDiver1975 committed:
Kill APC and XCache - fixes #28597

PVince81 commented:
bäh...
```
1) Test\Files\ViewTest::testLockBasicOperationUnlocksAfterException with data set #17 ('file_exists', array('dir'), 'dir', null)
File got unlocked after exception
Failed asserting that 1 is null.

/drone/src/tests/lib/Files/ViewTest.php:2073
```

individual-it commented:
@DeepDiver1975  #30483 should fix your drone issues

DeepDiver1975 commented:
@ownclouders rebase

ownclouders commented:
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

ownclouders commented:
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

DeepDiver1975 committed:
The tearDown call order can have an influence of the unit tests execution ....

DeepDiver1975 committed:
Add drone step to print the server logs in case of failure

DeepDiver1975 committed:
Properly mock the operation under test

phil-davis commented on .travis.yml:
```diff
@@ -56,364 +56,364 @@ script:
+    - php: 7.1
       env: DB=sqlite TC=syntax TEST_DAV=0
-    - php: 7.0
+    - php: 7.1
```
now duplicates the matrix entry above

phil-davis reviewed (commented):
Looks good - I only found 1 duplicated CI job.

phil-davis reviewed (approved):
All good as far as I can see. 
Of course there will be plenty of use/testing/manual QA to do in addition to the automated CI, to see if there are any problems with the various dependencies that bump upwards. But in theory that should all be better :)

PVince81 commented:
@phil-davis next up the Guzzle PR ?

phil-davis commented:
I will rebase it...

phil-davis commented:
Note: many parts of this PR are already back in `stable10` because they were not all directly just PHP7.1 commits.
The remaining PHP7.1-specific parts have been cherry-picked into backport PR #34936 ready for the day that we can drop PHP7.0 support. When those backport commits have been merged, then the `only-in-master` label can be removed here.
